frame/balances/e2e-tests/src/lib.rs

```diff
@@ -0,0 +1,148 @@
+use substrate_test_runner::{Node, ChainInfo};
+use pallet_sudo::Call as SudoCall;
+use pallet_balances::Call as BalancesCall;
+use sp_keyring::sr25519::Keyring::{Alice, Bob};
+use sp_runtime::{traits::IdentifyAccount, MultiSigner, MultiAddress, MultiSignature};
+use sp_runtime::AccountId32;
+use sp_runtime::traits::Block as BlockT;
+use sp_runtime::generic::UncheckedExtrinsic;
+use codec::Encode;
+use frame_support::traits::Get;
+
+type Extrinsic<R, S> = UncheckedExtrinsic<
+    MultiAddress<<R as frame_system::Config>::AccountId, <R as frame_system::Config>::Index>,
+    <R as frame_system::Config>::Call,
+    MultiSignature,
+    S,
+>;
+
+/// Tests the Balances::force_transfer call
+pub fn force_transfer<T>(node: &Node<T>)
+    where
+        T: ChainInfo,
+        T::Runtime: pallet_sudo::Config + pallet_balances::Config,
+        <T::Runtime as frame_system::Config>::Call: From<SudoCall<T::Runtime>> + From<BalancesCall<T::Runtime>> + Encode,
+        <T::Runtime as pallet_sudo::Config>::Call: From<BalancesCall<T::Runtime>>,
+        <T::Runtime as pallet_balances::Config>::Balance: From<u128>,
```

Comment:
u128 is a pretty high need, but the numbers used are in the low 10th. The from (here and elsewhere) should probably be bound to smallest needed value to allow for many cases to work. If I see this correctly a `From<u8>` would even be enough.

Reply:
u8 could work but it would be below existential deposit, u64 should be more general imo.

Reply:
> u8 could work but it would be below existential deposit

Isn't that an assumption? I _could_ configure my chain to have _no_ existential deposit, right? - e.g. there is a different mechanism to clean up the balances or that chain has a finite number of accounts anyways. 

We should not make any assumptions that limit otherwise valid use cases we just don't know about. The `From` here only needs to define what we need to be able to input. Why not go for the lowest value we can support but make other implied expectations? There is no harm in us doing that.

Otherwise we must make those assumptions explicit, by stating in the docs, that e.g. the tests only works if conditions X, Y, Z are met and why the specific test requires them to be met. E.g. if we were to test huge-transfers, we'd make clear that because we use numbers higher than MAX_INT, this only works on chains that can convert from `u64`.